playbooks/ansible-tox-py39/pre.yaml

```diff
@@ -1,8 +1,21 @@
 ---
 - hosts: all
   tasks:
-    - name: Ensure python3.9 is present
+
+    - name: Load operating system specific variables
+      include_vars: "{{ item }}"
+      failed_when: false
+      loop:
+        - "{{ ansible_os_family }}.yml"
+        - "{{ ansible_distribution | replace(' ', '-') }}.yml"
+        - "{{ ansible_distribution | replace(' ', '-') }}-{{ ansible_distribution_major_version }}.yml"
+        - "{{ ansible_distribution | replace(' ', '-') }}-{{ ansible_distribution_version.split('.')[0:2] | join('-') }}.yml"
+        - "{{ ansible_distribution | replace(' ', '-') }}-{{ ansible_distribution_version.split('.')[0:3] | join('-') }}.yml"
+      tags:
+        - always
+
+    - name: Ensure python3.9 devel is present
       become: true
       package:
-        name: python39-devel
+        name: "{{ python3_devel_pkg | default('python39-devel') }}"
```

Comment:
Honestly, this is overly complex and makes is much more difficult to review.  Over using 2 different set_fact statements. I'd like to see us use set_fact and default to python3.9, then check for fedora 34 and change the fact to python39-devel.

Then a single task to install like you have here.

We should make it pretty easier for reviewers here, over optimal code, given the lack of people reviewing code.

Reply:
doen